web/src/app/components/map/map.component.ts

```diff
@@ -497,17 +476,21 @@ export class MapComponent implements AfterViewInit, OnDestroy {
     polygon.addListener('click', (event: google.maps.PolyMouseEvent) => {
       this.onPolygonClick(event);
     });
-    this.polygons.set(loiId, polygon);
+    return polygon;
   }
 
   private onPolygonClick(event: google.maps.PolyMouseEvent) {
     if (this.disableMapClicks) {
       return;
     }
     const candidatePolygons: google.maps.Polygon[] = [];
-    for (const polygon of this.polygons.values()) {
-      if (google.maps.geometry.poly.containsLocation(event.latLng!, polygon)) {
-        candidatePolygons.push(polygon);
+    for (const loiPolygons of this.polygons.values()) {
+      for (const polygon of loiPolygons) {
```

Comment:
I can see one very specific corner case here. When an LOI is Multipolygon which has overlapping polygons, and user clicks into the intersection, then we would have 2 polygon candidates, but it's the same LOI, this would mess up the assumption of 1 polygon == 1 LOI by the code later.

I don't think we have to address this now and it's a very rare case, but can you add a TODO or comment to describe this situation?

Reply:
Ah yeah, true. Added a todo.